includes/class-wp-job-manager-email-notifications.php

```diff
@@ -0,0 +1,334 @@
+				|| ! isset( $email_notifications[ $email[0] ] )
+			) {
+				continue;
+			}
+
+			$class_name = $email_notifications[ $email[0] ];
+			$email_args = is_array( $email[1] ) ? $email[1] : array();
+
+			self::send_email( $email[0], new $class_name( $email_args ) );
+		}
+	}
+
+	/**
+	 * Include email files.
+	 *
+	 * Do not call manually.
+	 *
+	 * @access private
+	 */
+	public static function _lazy_init() {
+		add_action( 'shutdown', array( __CLASS__, '_send_deferred_notifications' ) );
+
+		include_once JOB_MANAGER_PLUGIN_DIR . '/includes/emails/class-wp-job-manager-email-admin-new-job.php';
+	}
+
+	/**
+	 * Clear the deferred notifications email array.
+	 *
+	 * Do not call manually. Only for help with tests.
+	 *
+	 * @access private
+	 */
+	public static function _clear_deferred_notifications() {
+		if ( ! defined( 'PHPUNIT_WPJM_TESTSUITE' ) || ! PHPUNIT_WPJM_TESTSUITE ) {
+			die( "This is just for use while testing" );
+		}
+		self::$deferred_notifications = array();
+	}
+
+	/**
+	 * Gets a list of all email notifications that WP Job Manager handles.
+	 *
+	 * @param bool $enabled_notifications_only
+	 * @return array
+	 */
+	public static function get_email_notifications( $enabled_notifications_only = false ) {
+		/**
+		 * Retrieves all email notifications to be sent.
+		 *
+		 * @since 1.31.0
+		 *
+		 * @param array $email_notifications All the email notifications to be registered.
+		 */
+		$email_notification_classes = array_unique( apply_filters( 'job_manager_email_notifications', self::core_email_notifications() ) );
+		$email_notifications = array();
+
+		/**
+		 * @var WP_Job_Manager_Email $email_class
+		 */
+		foreach ( $email_notification_classes as $email_class ) {
+			// Check to make sure email notification is valid.
+			if ( ! self::is_email_notification_valid( $email_class ) ) {
+				continue;
+			}
+
+			// PHP 5.2: Using `call_user_func()` but `$email_class::get_key()` preferred.
+			$email_notification_key = call_user_func( array( $email_class, 'get_key') );
+			if (
+				isset( $email_notifications[ $email_notification_key ] )
+				|| ( $enabled_notifications_only && ! self::is_email_notification_enabled( $email_notification_key ) )
+			) {
+				continue;
+			}
+
+			$email_notifications[ $email_notification_key ] = $email_class;
+		}
+
+		return $email_notifications;
+	}
+
+	/**
+	 * Generate the file name for the email template.
+	 *
+	 * @param string $template_name
+	 * @param bool   $plain_text
+	 * @return string
+	 */
+	public static function get_template_file_name( $template_name, $plain_text = false ) {
+		$file_name_parts = array( 'emails' );
+		if ( $plain_text ) {
+			$file_name_parts[] = 'plain';
+		}
+
+		$file_name_parts[] = $template_name . '.php';
+
+		return implode( '/', $file_name_parts );
+	}
+
+	/**
+	 * Returns the total number of deferred notifications to be sent.
+	 *
+	 * @access private
+	 *
+	 * @return int
+	 */
+	public static function _get_deferred_notification_count() {
```

Comment:
It looks like this function is only used by the unit tests. Should we mention that in the doc comment?

Reply:
Done 4082f95acda790eedb26fc9678548db8161e4721